### PR TITLE
Dockerfile for PPID retrieval tool

### PR DESCRIPTION
**Pull request by nshyrei**
This PR adds a Dockerfile for `PPID Retrieval` tool. Dockerfile includes all the necessary dependencies to `build` and `run` the tool. To make image modular I have used multi-stage build docker functionality which makes it easier to extract stages into a separate images if the need arises.

### CODE REVIEW

File: intel-sgx/ppid-decryption-tool/Dockerfile
Comment (Taowyoo) [inline intel-sgx/ppid-decryption-tool/Dockerfile:1]: Could you add a GitHub action to test this?
Here is template: https://github.com/fortanix/rust-sgx/new/master?filename=.github%2Fworkflows%2Fdocker-image.yml&workflow_template=ci%2Fdocker-image

I suggest you to just add it as separate action and only triggers by modification in folder `intel-sgx/ppid-decryption-tool` and github action yml file it self.
```
on:
  push:
    branches: ["master"]
    paths:
      - "intel-sgx/ppid-decryption-tool/**"
      - ".github/workflows/**"
  pull_request:
    branches: ["master"]
    paths:
      - "intel-sgx/ppid-decryption-tool/**"
      - ".github/workflows/**"
```

File: intel-sgx/ppid-decryption-tool/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+RUN DEBIAN_FRONTEND="noninteractive" TZ="America/New_York" apt-get install -y git build-essential python
+
+# Install user
+RUN useradd -rm -d /home/ppid-tool -s /bin/bash -g root -G sudo -u 1000 ppid-tool
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+RUN echo 'ppid-tool:ppid-tool' | chpasswd
+USER ppid-tool
+WORKDIR /home/ppid-tool
+
+# Install SGX Dev tools
```
Comment (Taowyoo) [inline intel-sgx/ppid-decryption-tool/Dockerfile:13]: Is it necessary to build & install sgx sdk from source?
Could we just install the dev pkgs like `libsgx-dcap-ql`
Note: I lack knowledge of how build works here.

File: intel-sgx/ppid-decryption-tool/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+RUN apt-get install -y libsgx-urts libsgx-epid libsgx-quote-ex libsgx-dcap-ql
+
+# Setup aesmd service in container
+USER root
+RUN bash -c 'mkdir /var/run/aesmd'
+RUN bash -c 'chown -c ppid-tool /var/run/aesmd'
+
+# Preparing to run PCKIDRetrievalTool
+USER root
+RUN bash -c 'apt-get install -y screen'
```
Comment (Taowyoo) [inline intel-sgx/ppid-decryption-tool/Dockerfile:86]: why we need `screen`?
```suggestion
```

Comment (nshyrei) [reply intel-sgx/ppid-decryption-tool/Dockerfile:86]: That I copied from an image created by @raoulstrackx , maybe he can answer.

File: intel-sgx/ppid-decryption-tool/Dockerfile (hunk 1 above)
Comment (nshyrei) [reply intel-sgx/ppid-decryption-tool/Dockerfile:13]: sdk env is needed to build the project

File: .github/workflows/build-docker-images.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Build the Docker image
+        run: |
+          cd intel-sgx/ppid-retrieval-tool
+          docker build -t ppid-retrieval-tool:$(date +%s) .
```
Comment (Taowyoo) [inline .github/workflows/build-docker-images.yml:28]: nit: missing last empty line

File: .github/workflows/build-docker-images.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# This workflow is used to build docker images
+
+name: Docker Image CI
```
Comment (Taowyoo) [inline .github/workflows/build-docker-images.yml:3]:
```suggestion
name: Docker Image CI for ppid-retrieval-tool
```
nit: suggest to have a more specific name

**Review by Taowyoo — DISMISSED** [review]

File: intel-sgx/ppid-decryption-tool/Dockerfile (hunk 2 above)
Comment (Taowyoo) [reply intel-sgx/ppid-decryption-tool/Dockerfile:86]: I mean `screen` is a tool similar to tmux. It's meaningless to install it in a docker image just for CI

File: intel-sgx/ppid-decryption-tool/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,96 @@
+From ubuntu:focal as sgx_sdk
```
Comment (raoulstrackx) [inline intel-sgx/ppid-decryption-tool/Dockerfile:1]: `From` needs to be `FROM`

File: intel-sgx/ppid-decryption-tool/Dockerfile (hunk 1 above)
Comment (raoulstrackx) [reply intel-sgx/ppid-decryption-tool/Dockerfile:13]: Most of the tools installed here, isn't needed. I think [this](https://github.com/intel/linux-sgx/blob/main/linux/installer/docker/Dockerfile#L32-L48) is only what's needed from Intel (but you need to update the distro used).

File: intel-sgx/ppid-decryption-tool/Dockerfile (hunk 5 above)
Comment (raoulstrackx) [inline intel-sgx/ppid-decryption-tool/Dockerfile:1]: Can you change this to ubuntu24?

**Review by raoulstrackx — CHANGES_REQUESTED** [review]

File: intel-sgx/ppid-retrieval-tool/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+RUN wget https://download.01.org/intel-sgx/sgx-linux/2.25/distro/ubuntu24.04-server/sgx_linux_x64_sdk_2.25.100.3.bin
+RUN chmod +x sgx_linux_x64_sdk_2.25.100.3.bin
+RUN echo 'yes' | ./sgx_linux_x64_sdk_2.25.100.3.bin
+
+# Install SGX runtime libraries
+USER root
+RUN echo 'deb [signed-by=/etc/apt/keyrings/intel-sgx-keyring.asc arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu noble main' | sudo tee /etc/apt/sources.list.d/intel-sgx.list
+RUN wget https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key
+RUN cat intel-sgx-deb.key | sudo tee /etc/apt/keyrings/intel-sgx-keyring.asc > /dev/null
+RUN apt-get install -y libsgx-urts
```
Comment (raoulstrackx) [inline intel-sgx/ppid-retrieval-tool/Dockerfile:31]: You removed one `apt-get update` too many

**Review by raoulstrackx — APPROVED** [review]

**Review by raoulstrackx — APPROVED** [review]

**Pull request merged**